docs/howto/krustlet-on-microk8s.md

```diff
@@ -0,0 +1,141 @@
+```
+
+This will create a CSR and a new key for the certificate, using `krustlet` as the hostname of the
+server.
+
+Now that it is created, we'll need to send the request to Kubernetes:
+
+```shell
+$ cat <<EOF | microk8s.kubectl apply --filename=-
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: krustlet
+spec:
+  request: $(cat krustlet.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+EOF
+certificatesigningrequest.certificates.k8s.io/krustlet created
+```
+
+Once that runs, an admin (that is probably you! at least it should be if you are trying to add a
+node to the cluster) needs to approve the request:
+
+```shell
+$ microk8s.kubectl certificate approve krustlet
+certificatesigningrequest.certificates.k8s.io/krustlet approved
+```
+
+After approval, you can download the cert like so:
+
+```shell
+$ microk8s.kubectl get csr krustlet --output=jsonpath='{.status.certificate}' \
+    | base64 --decode > krustlet.crt
+```
+
+Lastly, combine the key and the cert into a PFX bundle, choosing your own password instead of
+"password":
+
+```shell
+$ openssl pkcs12 -export -out krustlet.pfx -inkey krustlet.key -in krustlet.crt -password "pass:password"
+```
+
+## Step 3: Install and configure Krustlet
+
+Install the latest release of krustlet following [the install guide](../intro/install.md).
+
+We want the krustlet to run as the service account that we created in step #1. This is configured by the context (`krustlet`) that we created in that step. Unfortunately, it's not possible to reference a specific context, so we must change the context before running krustlet:
+
+```shell
+$ microk8s.kubectl use context ${CONTEXT}
+Switched to context "krustlet".
+```
+
+There are 2 binaries (`krustlet-wasi` and `krustlet-wascc`), let's start the first:
+
+```shell
+$ KUBECONFIG=/var/snap/microk8s/current/credentials/client.config \
+./krustlet-wasi \
+--node-ip=127.0.0.1 \
+--node-name=krustlet \
+--pfx-password="password" \
+--pfx-path=./krustlet.pfx
+```
+
+In another terminal:
+
+```shell
+$ microk8s.kubectl get nodes --output=wide
+NAME                                STATUS   ROLES   AGE     VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
+krustlet                            Ready    agent   11s     v1.17.0   127.0.0.1     <none>        <unknown>            <unknown>           mvp
+microk8s                            Ready    <none>  13m     v1.18.2   10.138.0.4    <none>        Ubuntu 20.04 LTS     5.4.0-1009-gcp      containerd://1.2.5
+```
+
+## Step 6: Test that things work
+
+Now you can see things work! Feel free to give any of the demos a try like so:
+
+```shell
+$ microk8s.kubectl apply --file=https://raw.githubusercontent.com/deislabs/krustlet/master/demos/wasi/hello-world-rust/k8s.yaml
```

Comment:
Does this need a context switch back after having switched it to `krustlet` above?

Reply:
The instructions run Krustlet in the foreground.

I didn't include the section on running it as a systemd service.

As with the `kubectl get nodes ...` command, this would need to be run from a different shell.

Reply:
You had this above:
```
$ microk8s.kubectl use context ${CONTEXT}
 Switched to context "krustlet".
```

Which means that it would be set in any shell. I was just double checking that this worked as intended. By no means a blocker

Reply:
Right! Will change.

Reply:
Fixed 120e4b05d4857bd0b10a8ccfb564e5d6569986dd